oio/api/object_storage.py

```diff
@@ -823,6 +823,49 @@ def object_analyze(self, *args, **kwargs):
                       DeprecationWarning)
         return self.object_locate(*args, **kwargs)
 
+    def _generate_fullchunk_copies(self, chunks, random_hex=60):
+        # random_hex is the number of hexadecimals characters to generate for
+        # the copy path
+        copies = []
+        for c in chunks:
+            tmp = c.copy()
+            tmp['url'] = ''.join([tmp['url'][:-random_hex],
+                           ''.join(random.choice('0123456789ABCDEF') for _ in range(random_hex))])
+            copies.append(tmp)
+        return copies
+
+    def object_fastcopy(self, target_account, target_container, target_obj,
+                        link_account, link_container, link_obj,
+                        version=None,  **kwargs):
+        meta, chunks = self.object_locate(
+            target_account, target_container, target_obj, version=version, **kwargs)
+
+        chunks_copies = self._generate_fullchunk_copies(chunks)
+        chunks_copies_url = []
+        chunks_url= []
+        for c in chunks:
+            chunks_url.append(c['url'])
+        for c in chunks_copies:
+            chunks_copies_url.append(c['url'])
+        fullpath = self._generate_fullpath(link_account, link_container, link_obj,
+                                           version)
+        self._send_copy(chunks_url, chunks_copies_url, fullpath[0])
+        data = {'chunks': chunks_copies,
+                'properties': meta["properties"] or {}}
+        self.container.content_create(link_account, link_container, link_obj,
+                                      size=meta["length"], checksum=meta['hash'],
+                                      data=data,
+                                      stgpol=meta["policy"],version=meta['version'], mime_type=meta['mime_type'],
+                                      chunk_method=meta['chunk_method'], **kwargs)
+
+    def _send_copy(self, targets, copies, fullpath):
+        headers = {"x-oio-chunk-meta-full-path": fullpath}
+        if not hasattr(self, "blob_client"):
+            from oio.blob.client import BlobClient
+            self.blob_client = BlobClient()
```

Comment:
You can use `self.blob_client` directly. It's a [`property`](#diff-0f6dcfc8155e770f4df2e954d7976e04R355).